pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -150,9 +150,9 @@ func filter(context context.Context, schedulerExtenderArgs schedulerapi.Extender
 	ns := schedulerExtenderArgs.Pod.Namespace
 
 	// get requested disks created by Azure Disk V2 driver and return early if none are found
-	diskRequestedByPod, totalRequestedDiskCount := getQualifyingDisksRequestedByPod(ns, requestedVolumes)
-	// TODO: change to len(diskRequestedByPod) when we fix #1097 in the controller
-	if len(requestedVolumes) == 0 {
+	diskRequestedByPod, _ := getQualifyingDisksRequestedByPod(ns, requestedVolumes)
+	// when there is no v2 disks requested by pod, don't filter and return
```

Comment:
`when there are no v2 disks requested by the pod, skip filtering and return all nodes.`